### ISSUE TITLE
Smaller Models?

### ISSUE THREAD

**Issue by jgolliher** [AnswerDotAI/ModernBERT#170]
First off- this is fantastic! I'm using BERT in many applications and was always wondering if encoder-only models would get any love.

I'm curious if there's any interest (or need) to release smaller variants of ModernBERT similar to what Google [did](https://github.com/google-research/bert?tab=readme-ov-file#bert) with the original BERT model. One of my tasks works really well with BERT-small (4/512) but initial training times with ModernBERT on a 16GB V100 are quite long (i.e., BERT-small takes about 10 minutes/epoch while ModernBERT estimates over 12 hours an epoch). While I'm sure there's more optimization to be done on my end, it made me wonder if there's any interest in releasing smaller models. 

Thanks!

**Comment by NohTow** [comment]
Hello,
Thank you!

For the time being, there is no plan of training smaller models, actually the planned exploration is more on the opposite side of the spectrum: scaling a bit the model to see what is the benefit of scaling encoders.
I suspect there will be some distilled version available (e.g, this [model](https://huggingface.co/andersonbcdefg/distilmodernbert)), which might fit your needs, but not smaller pre-trained models (at least for now).

About performance with the existing models, did you install flash-attention? Could bring a cool speed-up!

**Comment by jgolliher** [comment]
Thank you for the response! I figured others might create a distilled version, so I'll keep an eye out for it! 

Truthfully, I'm still working in TensorFlow (which was pretty standard when I was in school, but the times have changed)! From what I can tell, flash-attention only supports PyTorch (and possibly JAX) at the moment. Might be a good opportunity for me to learn PyTorch.

**Comment by BlessedTatonka** [comment]
@NohTow
What argument values would you suggest for smaller versions? Are there any insights? Any advice would be helpful.

**Comment by tjasmin111** [comment]
Following. I'm also interested in a smaller model. Likely a ModernBERT variation of <50M parameters.
Please let me know if you found anything.

**Comment by BlessedTatonka** [comment]
@tjasmin111 
Sure, we have already trained such a model, you can check it [here](https://huggingface.co/deepvk/RuModernBERT-small). All training anrguments are available at config.json file.

**Comment by tjasmin111** [comment]
@BlessedTatonka  35M sounds great. What does it mean "The Russian version"? Is it only working on Russian language?

**Comment by tjasmin111** [comment]
Also, I see comparisons against deberta-v1. Have you evaluated it against Deberta-v3?

**Comment by BlessedTatonka** [comment]
Actually, I’m part of lab, which works mostly on Russian focused models. However, this one was trained on a mix of both Russian and English (~1T tokens of each).

You can try to validate accuracy on your task, I believe, it should be competitive with the same size encoders (ex. deberta-small). Also, you can use it as a backbone for continuous pre-training, and lower its’ size even more by tokenizer shrinking. Hope it helps.

Upd. We did evaluate model on Russian bench, I believe those charts are not appropriate for your task. On English we did some sanity check and model worked good.
